doc/designs/quic-design/quic-event-loop.md

```diff
@@ -0,0 +1,160 @@
+# Design Problem: Event Loop
+
+Applications need to be able to react to events on:
+
+* Sockets
+* Connections
+* Streams
+* Timers
+
+Only the OpenSSL library can supply the events for Connections and Streams.
+Events on sockets and timers ultimately come from a system source - the
+application may want to receive the event directly from the system, or from
+OpenSSL.
+
+For the MVP a blocking mode on top of one of the options below is likely to be
+required because that’s the default mode for `s_client`.
```

Comment:
Note that s_client uses select and supports non-blocking IO usage - so working with applications that handle their own select is a requirement for the MVP.